borg/helpers.py

```diff
@@ -257,21 +257,33 @@ def update_excludes(args):
             file.close()
 
 
-def adjust_patterns(paths, excludes):
-    if paths:
-        return (excludes or []) + [PathPrefixPattern(path) for path in paths] + [FnmatchPattern('*')]
-    else:
-        return excludes
+class PatternMatcher:
+    def __init__(self):
+        self._items = []
+        self._fallback = None
 
+    @property
+    def fallback(self):
+        """Value to return from match function when none of the patterns match.
+        """
+        return self._fallback
 
-def exclude_path(path, patterns):
-    """Used by create and extract sub-commands to determine
-    whether or not an item should be processed.
-    """
-    for pattern in (patterns or []):
-        if pattern.match(path):
-            return isinstance(pattern, (FnmatchPattern, RegexPattern))
-    return False
+    @fallback.setter
+    def fallback(self, value):
+        self._fallback = bool(value)
+
+    def add(self, patterns, value):
+        """Add list of patterns to internal list. The given value is returned from the match function when one of the
+        given patterns matches.
+        """
```

Comment:
Considering that the matcher can work with any type and value (after the removal of that explicit conversion to bool which will be in the next push), adding that would not be helpful. If you look at the test you see that it works with strings.